From 02ff3023104220c3041d39fb14222df67ba7de79 Mon Sep 17 00:00:00 2001
From: "guoyin.chen" <guoyin.chen@freescale.com>
Date: Mon, 7 Mar 2016 15:12:40 +0800
Subject: [PATCH 8/8] MA-7695 testJobParameters_expiredDeadline failed on DL
 ARD

ARD board boot will have below exception, which will block the Broadcast intent 60s
from system server.
jobscheduler service depends on the AlarmManagerService to broadcast the intent to
CTS on the JOB_DEADLINE_EXPIRED job. As those CTS cases is start just after the boot
process. The CTS cannot get the JOB_DEADLINE_EXPIRED job's intent in time

Caused by: java.lang.RuntimeException: Fail to get camera info
    at android.hardware.Camera._getCameraInfo(Native Method)
    at android.hardware.Camera.getCameraInfo(Camera.java:236)
    at com.android.camera.DisableCameraReceiver.hasBackCamera(DisableCameraReceiver.java:65)
    at com.android.camera.DisableCameraReceiver.onReceive(DisableCameraReceiver.java:41)
    at android.app.ActivityThread.handleReceiver(ActivityThread.java:2725)

Change-Id: I09bdd91f0ae8bf71f25ae0bdca97962d5e6724cb
Signed-off-by: guoyin.chen <guoyin.chen@freescale.com>
---
 src/com/android/camera/DisableCameraReceiver.java | 16 +++++++++++-----
 1 file changed, 11 insertions(+), 5 deletions(-)

diff --git a/src/com/android/camera/DisableCameraReceiver.java b/src/com/android/camera/DisableCameraReceiver.java
index aea6ec5..e7bab1e 100644
--- a/src/com/android/camera/DisableCameraReceiver.java
+++ b/src/com/android/camera/DisableCameraReceiver.java
@@ -62,13 +62,19 @@ public class DisableCameraReceiver extends BroadcastReceiver {
     private boolean hasBackCamera() {
         int n = android.hardware.Camera.getNumberOfCameras();
         CameraInfo info = new CameraInfo();
-        for (int i = 0; i < n; i++) {
-            android.hardware.Camera.getCameraInfo(i, info);
-            if (info.facing == CameraInfo.CAMERA_FACING_BACK) {
-                Log.i(TAG, "back camera found: " + i);
-                return true;
+
+        try {
+            for (int i = 0; i < n; i++) {
+                android.hardware.Camera.getCameraInfo(i, info);
+                if (info.facing == CameraInfo.CAMERA_FACING_BACK) {
+                    Log.i(TAG, "back camera found: " + i);
+                    return true;
+                }
             }
+        } catch (Exception e) {
+            Log.e(TAG, "Camera Info is unavailable for queries");
         }
+
         Log.i(TAG, "no back camera");
         return false;
     }
-- 
1.8.0

